packages/python/plotly/plotly/express/_core.py

```diff
@@ -754,6 +754,135 @@ def apply_default_cascade(args):
+        for i, (argument, field) in enumerate(zip(argument_list, field_list)):
+            length = len(df)
+            if argument is None:
+                continue
+            elif isinstance(argument, str) or isinstance(
+                argument, int
+            ):  # just a column name
+                if not isinstance(
+                    args.get("data_frame"), pd.DataFrame
+                ) and not isinstance(args.get("data_frame"), np.ndarray):
+                    raise ValueError(
+                        "String or int arguments are only possible when a"
+                        "DataFrame or an array is provided in the `data_frame`"
+                        "argument."
+                    )
+                # Check validity of column name
+                if df_columns is not None and argument not in df_columns:
+                    raise ValueError(
+                        "Value of '%s' is not the name of a column in 'data_frame'. "
+                        "Expected one of %s but received: %s"
+                        % (field, str(list(df_columns)), argument)
+                    )
+                if length and len(args["data_frame"][argument]) != length:
+                    raise ValueError("All arguments should have the same length.")
+                df[str(argument)] = args["data_frame"][argument]
+                if isinstance(argument, int):
+                    if field_name not in array_attrables:
+                        args[field_name] = str(argument)
+                    else:
+                        args[field_name][i] = str(argument)
+                continue
+            # Case of index
+            elif isinstance(argument, pd.core.indexes.multi.MultiIndex):
+                raise TypeError("pandas MultiIndex is not supported by plotly express")
+            # Case of numpy array or df column
+            else:
+                try:
+                    col_name = argument.name  # pandas df
+                    if col_name is None and isinstance(
+                        argument, pd.core.indexes.range.RangeIndex
+                    ):
+                        col_name = "index"
+                    if (
+                        args.get("data_frame") is not None
+                        and col_name in args["data_frame"]
+                    ):
+                        # If the name exists but the values have changed
+                        # we do not want to keep the name, revert to field
+                        col_name = (
+                            col_name
+                            if argument is args["data_frame"][col_name]
+                            else field
+                        )
+                except AttributeError:  # numpy array, list...
+                    col_name = field
+                if length and len(argument) != length:
+                    raise ValueError("All arguments should have the same length.")
```

Comment:
we should indicate which attribute is causing the issue here (I know in case of length it's ambiguous, but any hint helps!)

Reply:
maybe report the previous argument whose length we compared to?